src/library_manager/lib_manager.c

```diff
@@ -451,8 +453,10 @@ static int lib_manager_dma_init(struct lib_manager_dma_ext *dma_ext, uint32_t dm
 
 static int lib_manager_dma_deinit(struct lib_manager_dma_ext *dma_ext, uint32_t dma_id)
 {
-	dma_release_channel(dma_ext->dma->z_dev, dma_id);
-	dma_put(dma_ext->dma);
+	if (dma_ext->dma->z_dev)
+		dma_release_channel(dma_ext->dma->z_dev, dma_id);
+	if (dma_ext->dma)
+		dma_put(dma_ext->dma);
```

Comment:
the ordering of the if's above don't quite add up right? if (dma_ext->dma) is false, how can we check for if (dma_ext->dma->z_dev) first? And also, you talk about init issue in the commit message and this change isnt quite related to that at all

Reply:
It is true of course. Fixed.